src/librustc/middle/const_eval.rs

```diff
@@ -90,7 +91,8 @@ fn lookup_variant_by_id<'a>(tcx: &'a ty::ctxt,
 
 pub fn lookup_const_by_id<'a, 'tcx: 'a>(tcx: &'a ty::ctxt<'tcx>,
                                         def_id: DefId,
-                                        maybe_ref_id: Option<ast::NodeId>)
+                                        maybe_ref_id: Option<ast::NodeId>,
+                                        param_substs: Option<&'tcx subst::Substs<'tcx>>)
```

Comment:
As an aside -- this all feels very hokey to me, I'd rather we just passed in the final set of substitutions, rather than passing in an expr-id and monomorphization substitution and then figuring out the substitutions. But that feels like a separate PR.